hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestAbfsCustomTimeout.java

```diff
@@ -0,0 +1,155 @@
+            }
+            else if (opType == AbfsRestOperationType.DeleteFileSystem) {
+                config = ConfigurationKeys.AZURE_DELETE_FS_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.ListPaths) {
+                config = ConfigurationKeys.AZURE_LIST_PATH_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.CreatePath) {
+                config = ConfigurationKeys.AZURE_CREATE_PATH_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.RenamePath) {
+                config = ConfigurationKeys.AZURE_RENAME_PATH_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.GetAcl) {
+                config = ConfigurationKeys.AZURE_GET_ACL_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.GetPathProperties) {
+                config = ConfigurationKeys.AZURE_GET_PATH_PROPERTIES_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.SetPathProperties) {
+                config = ConfigurationKeys.AZURE_SET_PATH_PROPERTIES_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.SetAcl) {
+                config = ConfigurationKeys.AZURE_SET_ACL_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.SetOwner) {
+                config = ConfigurationKeys.AZURE_SET_OWNER_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.SetPermissions) {
+                config = ConfigurationKeys.AZURE_SET_PERMISSIONS_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.CheckAccess) {
+                config = ConfigurationKeys.AZURE_CHECK_ACCESS_REQUEST_TIMEOUT;
+            }
+            else if (opType == AbfsRestOperationType.GetPathStatus) {
+                config = ConfigurationKeys.AZURE_GET_PATH_STATUS_REQUEST_TIMEOUT;
+            }
+            abfsConfig.set(config, Integer.toString(timeout));
+            testInitTimeoutOptimizer(opType, 3, timeout, newConfig);
+            abfsConfig.unset(config);
+        }
+
+        abfsConfig.set(ConfigurationKeys.AZURE_OPTIMIZE_TIMEOUTS, "false");
+
+    }
+
+    public void testInitTimeoutOptimizer(AbfsRestOperationType opType, int maxRetryCount, int expectedReqTimeout, AbfsConfiguration abfsConfig) throws IOException {
+
+        AzureBlobFileSystem fs = (AzureBlobFileSystem) FileSystem.newInstance(abfsConfig.getRawConfiguration());
+        AbfsClient client = fs.getAbfsStore().getClient();
+        String query = client.createDefaultUriQueryBuilder().toString();
+        URL url = client.createRequestUrl("/testPath", query);
+        TimeoutOptimizer opt = new TimeoutOptimizer(url, opType, client.getRetryPolicy(), getConfiguration());
+        int retryCount = 0;
+        while (retryCount <= maxRetryCount) {
```

Comment:
Great test. This would be a unit-test for the functionality of TimeoutOptimizer.
Can we add a test where-in, processResponse() of abfsHttpOperation fails, and we assert how abfsRestOperation and TimeoutOptimizer is working.